src/tests/Parser/Brewmaster/Stagger.test.js

```diff
@@ -1,55 +1,68 @@
 import Stagger from 'Parser/Monk/Brewmaster/Modules/Core/Stagger';
 import processEvents from './Fixtures/processEvents';
 import { SimpleFight, EarlyFinish, incomingDamage } from './Fixtures/SimpleFight';
 
 describe('Brewmaster.Stagger', () => {
   let stagger;
+  let fab;
   beforeEach(() => {
+    fab = new StaggerFabricator({
+      toPlayer: () => true,
+      byPlayer: () => true,
+      toPlayerPet: () => false,
+      byPlayerPet: () => false,
+    });
+    fab.combatants = {
+      selected: {
+        hasTalent: () => false,
+        traitsBySpellId: {[SPELLS.STAGGERING_AROUND.id]: 0},
+      }
+    };
     stagger = new Stagger({
       toPlayer: () => true,
       byPlayer: () => true,
       toPlayerPet: () => false,
       byPlayerPet: () => false,
     });
+    stagger.fab = fab;
+    fab.owner.triggerEvent = (event, obj) => stagger.triggerEvent(event, obj);
   });
   it('total amount of stagger taken with no events', () => {
     expect(stagger.totalStaggerTaken).toBe(0);
   });
   it('total amount of stagger taken with no stagger', () => {
-    processEvents(incomingDamage, stagger);
+    processEvents(incomingDamage, fab, stagger);
     expect(stagger.totalStaggerTaken).toBe(0);
   });
   it('total amount of stagger damage taken', () => {
-    processEvents(SimpleFight, stagger);
+    processEvents(SimpleFight, fab, stagger);
     expect(stagger.totalStaggerTaken).toBe(240);
   });
   it('track how much damage in total is absorbed by stagger, with no stagger absorbs', () => {
-    processEvents(incomingDamage, stagger);
+    processEvents(incomingDamage, fab, stagger);
     expect(stagger.totalPhysicalStaggered + stagger.totalMagicalStaggered).toBe(0);
   });
   it('track how much damage in total is absorbed by stagger', () => {
-    processEvents(SimpleFight, stagger);
+    processEvents(SimpleFight, fab, stagger);
     expect(stagger.totalPhysicalStaggered + stagger.totalMagicalStaggered).toBe(599);
   });
   it('track how much physical damage was staggered', () => {
-    processEvents(SimpleFight, stagger);
+    processEvents(SimpleFight, fab, stagger);
     expect(stagger.totalPhysicalStaggered).toBe(300);
   });
   it('track how much magical damage was staggered', () => {
-    processEvents(SimpleFight, stagger);
+    processEvents(SimpleFight, fab, stagger);
     expect(stagger.totalMagicalStaggered).toBe(299);
   });
-  it('When was the last non stagger damage event', () => {
-    processEvents(SimpleFight, stagger);
-    expect(stagger.lastDamageEventNotStagger).toBe(5700);
-  });
   it('Tracks the amount of stagger missing from the fight', () => {
     const earlyFightEnd = 6000;
     const myOwner = { fight: { end_time: earlyFightEnd } };
-    processEvents(EarlyFinish, stagger);
+    processEvents(EarlyFinish, fab, stagger);
     stagger.owner = myOwner;
     stagger.triggerEvent('finished');
-    expect(stagger.staggerMissingFromFight).toBe(285);
+    expect(stagger.staggerMissingFromFight).toBe(484);
```

Comment:
The `staggerMissingFromFight` value *had been* calculated based on the most recent tick of stagger damage. The new method tracks the actual amount in the pool over the entirety of the fight. The stagger ticks in the fixture are not what they would be in an actual fight. This led to the old method underestimating the amount of damage missing *on the fixture*.

I spoke to WOPR about it via Discord PMs and his immediate response was that the test data was probably wrong, which lines up with my math. I noted this in my [previous comment](https://github.com/WoWAnalyzer/WoWAnalyzer/pull/1081#issuecomment-354019767).

The test data needs to be replaced, but I'm not totally sure what to do for it yet (dump a real log to a file?) because stagger is a complicated piece of shit.